### PR TITLE
calamares: fix locale errors and update

### LINKED ISSUE #11: Add more connectivity check URLs in case geoip.kde.org goes down
It's down right now and there's at least two people on Matrix unable to continue installing.

### LINKED ISSUE #180935: Error: unsupported locales detected
### Describe the bug

When using systemd-boot, installing NixOS or updating system using `nixos-rebuild` will fail at `glibc-locales` with error

```
Error: unsupported locales detected:
en_US.utf8/UTF8
```

### Steps To Reproduce
1. Download this [unstable version NixOS image](https://releases.nixos.org/nixos/unstable/nixos-22.11pre389925.a5c867d9fe9).
2. Use installer to install NixOS.
  - Location: Asia/Shanghai
  - Partitions: Erase entire disk and enable swap
3. The installation process will fail with message described in the description.

### Expected behavior

Can install NixOS or update system.

### Screenshots

Screenshot_20220710_144939

### Additional context

No

### Notify maintainers

Ping maintainers of `glibc-locales`.

@edolstra @Ma27 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.52, NixOS, 22.11 (Raccoon), 22.11pre389925.a5c867d9fe9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(root): `"nixos-22.11pre389925.a5c867d9fe9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by vlinkz**
###### Description of changes

Changes that fix the calamares installer on unstable, update versions, and add more stability by adding a backup network check url.

###### Things done

- Update calamares to 0.3.59
- Add SUPPORTED file to glibcLocales
- Make calamares read that SUPPORTED files to find suitable locales
  - fixes #180935
  - Caused as a side effect of #177318
- Add https://cache.nixos.org connectivity check
  - fixes https://github.com/NixOS/calamares-nixos-extensions/issues/11

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/installation-cd-graphical-calamares.nix [hunk 1]
```diff
@@ -16,5 +16,11 @@ in
     calamares-nixos-extensions
     # Needed for calamares QML module packagechooserq
     libsForQt5.full
+    # Get list of locales
+    glibcLocales
   ];
+
+  # Support choosing from any locale
+  i18n.supportedLocales = [ "all" ];
+
```
Comment (SuperSandro2000) [inline nixos/modules/installer/cd-dvd/installation-cd-graphical-calamares.nix:25]:
```suggestion
```

File: pkgs/development/libraries/glibc/locales.nix [hunk 2]
```diff
@@ -58,7 +58,9 @@ callPackage ./common.nix { inherit stdenv; } {
   installPhase =
     ''
       mkdir -p "$out/lib/locale"
+      mkdir -p "$out/share/i18n"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/glibc/locales.nix:61]:
```suggestion
      mkdir -p "$out/lib/locale" "$out/share/i18n"
```

**Comment by vlinkz** [comment]
Left this on the back burner for too long. Should fix issues with unstable ISO for now, but some major changes will be needed to the patches once calamares 3.3.x is released in a few months

**Comment by SuperSandro2000** [comment]
This PR rebuilds a lot of packages which means we must target staging. Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging) to not potentially ping a lot of people.

@ofborg eval

**Pull request closed without merging**